### ISSUE TITLE
Non-trainable parameters seems to have not work

### ISSUE THREAD

**Issue by CHRIS-LIN-Jude** [sksq96/pytorch-summary#55]
----------------------------------------------------------------
        Layer (type)               Output Shape         Param #
================================================================
    
```
            Conv2d-1          [1, 16, 240, 240]             160
       BatchNorm2d-2          [1, 16, 240, 240]               0
              ReLU-3          [1, 16, 240, 240]               0
            Conv2d-4          [1, 16, 240, 240]           2,320
       BatchNorm2d-5          [1, 16, 240, 240]               0
         MaxPool2d-6          [1, 16, 120, 120]               0
         Dropout2d-7          [1, 16, 120, 120]               0
          DownConv-8  [[-1, 16, 120, 120], [-1, 16, 240, 240]]               0
            Conv2d-9          [1, 32, 120, 120]           4,640
      BatchNorm2d-10          [1, 32, 120, 120]               0
             ReLU-11          [1, 32, 120, 120]               0
           Conv2d-12          [1, 32, 120, 120]           9,248
      BatchNorm2d-13          [1, 32, 120, 120]               0
        MaxPool2d-14            [1, 32, 60, 60]               0
        Dropout2d-15            [1, 32, 60, 60]               0
         DownConv-16  [[-1, 32, 60, 60], [-1, 32, 120, 120]]               0
           Conv2d-17            [1, 64, 60, 60]          18,496
      BatchNorm2d-18            [1, 64, 60, 60]               0
             ReLU-19            [1, 64, 60, 60]               0
           Conv2d-20            [1, 64, 60, 60]          36,928
      BatchNorm2d-21            [1, 64, 60, 60]               0
        MaxPool2d-22            [1, 64, 30, 30]               0
        Dropout2d-23            [1, 64, 30, 30]               0
         DownConv-24  [[-1, 64, 30, 30], [-1, 64, 60, 60]]               0
           Conv2d-25           [1, 128, 30, 30]          73,856
      BatchNorm2d-26           [1, 128, 30, 30]               0
             ReLU-27           [1, 128, 30, 30]               0
           Conv2d-28           [1, 128, 30, 30]         147,584
      BatchNorm2d-29           [1, 128, 30, 30]               0
        MaxPool2d-30           [1, 128, 15, 15]               0
        Dropout2d-31           [1, 128, 15, 15]               0
         DownConv-32  [[-1, 128, 15, 15], [-1, 128, 30, 30]]               0
           Conv2d-33           [1, 256, 15, 15]         295,168
      BatchNorm2d-34           [1, 256, 15, 15]               0
             ReLU-35           [1, 256, 15, 15]               0
           Conv2d-36           [1, 256, 15, 15]         590,080
      BatchNorm2d-37           [1, 256, 15, 15]               0
         DownConv-38  [[-1, 256, 15, 15], [-1, 256, 15, 15]]               0
  ConvTranspose2d-39           [1, 128, 30, 30]         131,200
        Dropout2d-40           [1, 256, 30, 30]               0
           Conv2d-41           [1, 128, 30, 30]         295,040
      BatchNorm2d-42           [1, 128, 30, 30]               0
           Conv2d-43           [1, 128, 30, 30]         147,584
      BatchNorm2d-44           [1, 128, 30, 30]               0
           UpConv-45           [1, 128, 30, 30]               0
  ConvTranspose2d-46            [1, 64, 60, 60]          32,832
        Dropout2d-47           [1, 128, 60, 60]               0
           Conv2d-48            [1, 64, 60, 60]          73,792
      BatchNorm2d-49            [1, 64, 60, 60]               0
           Conv2d-50            [1, 64, 60, 60]          36,928
      BatchNorm2d-51            [1, 64, 60, 60]               0
           UpConv-52            [1, 64, 60, 60]               0
  ConvTranspose2d-53          [1, 32, 120, 120]           8,224
        Dropout2d-54          [1, 64, 120, 120]               0
           Conv2d-55          [1, 32, 120, 120]          18,464
      BatchNorm2d-56          [1, 32, 120, 120]               0
           Conv2d-57          [1, 32, 120, 120]           9,248
      BatchNorm2d-58          [1, 32, 120, 120]               0
           UpConv-59          [1, 32, 120, 120]               0
  ConvTranspose2d-60          [1, 16, 240, 240]           2,064
        Dropout2d-61          [1, 32, 240, 240]               0
           Conv2d-62          [1, 16, 240, 240]           4,624
      BatchNorm2d-63          [1, 16, 240, 240]               0
           Conv2d-64          [1, 16, 240, 240]           2,320
      BatchNorm2d-65          [1, 16, 240, 240]               0
           UpConv-66          [1, 16, 240, 240]               0
           Conv2d-67           [1, 1, 240, 240]              17


================================================================
Total params: 1,940,817
Trainable params: 1,940,817
Non-trainable params: 0

----------------------------------------------------------------

Input size (MB): 0.22
Forward/backward pass size (MB): 2177055.62
Params size (MB): 7.40
Estimated Total Size (MB): 2177063.24

----------------------------------------------------------------
```

The BatchNorm2d I set affine is False, while there will some Non-trainable params in the Network,
while I do the same things in Keras, and I can't count the correct parameters number in training and estimated the batch size in the work,
What happed with it?
Thanks you,
Chris

**Comment by elvisyjlin** [comment]
In Batch Normalization, gamma=1 and beta=0 if affine is set to False. That is, it simply computes `y = (x - mu) / sqrt(var + eps)`, where mu is the running mean and var is the running variance.

There is no need for the weight and bias so they are set as None in PyTorch implementation. In Keras, they probably set as non-trainable variables whilst PyTorch doesn't create tensors for them. There is no way to compute the number of parameters of nothing.

See https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L15.

**Comment by Naireen** [comment]
Going to close this given the above comment, thanks for the clarification @elvisyjlin
